src/ceph-disk/ceph_disk/main.py

```diff
@@ -2364,7 +2364,8 @@ def prepare_device(self):
                 self.space_symlink = getattr(self.args, self.name)
                 return
 
-        self.space_symlink = '/dev/disk/by-partuuid/{uuid}'.format(
+        self.space_symlink = '/dev/disk/by-parttypeuuid/{ptype}.{uuid}'.format(
```

Comment:
@jmamma i don't think we can deduce that ptype is `PTYPE['mpath']['block']['ready']`. 1) why is it multipath device? 2) what if it is a "journal"? 3) why is it ready before it is created and marked using `sgdisk`?

Reply:
I'm not sure either why this change is a good thing. An explanation would be great :-)

Reply:
I've also encountered the problem this PR is trying to solve and cannot use ceph-disk to create bluestore devices (though I can fix the symlink manually after the creation).

If devices are multipath, they do not show up in /dev/disk/by-partuuid.  The multipath sub-devices are there (like /dev/sdda) but not the aggregated /dev/dm-XX you would want to point at for the ceph block storage device.  This is apparently by intent as dm- devices are excluded in udev rules.d/60-persistent-storage.rules where the by-partuuid links are created.

However ALL devices show up in by-parttypeuuid including dm- and nvme, etc devices.   So it seems like this would be the most appropriate place to link to.

I agree that assuming a ptype 'mpath' is incorrect.   At a glance it looks like simply using partition.get_ptype() to set ptype would work to ensure the right type is set for all circumstances.   Maybe the PR could be updated with that?  Or I can try the change here and put in another PR if it works as expected.

Reply:
This commit slightly modifying the commit from the PR works, I've tested it to correctly create the symlink.  The location of the space_symlink definition has to be moved a little to come after a partition exists:   https://github.com/MI-OSiRIS/ceph/commit/29c5445fb531eb8ec7b1218c8bd9ff680c36b8d3